lib/tapioca/dsl/compilers/active_record_relations.rb

```diff
@@ -560,12 +550,25 @@ def create_common_methods
                 return_type: "T::Boolean",
               )
             when :find
-              create_common_method(
+              args_type = if constant.try(:composite_primary_key?)
+                "T::Array[T::Array[T.untyped]]"
+              else
+                "T::Array[T.untyped]"
+              end
+              sigs = [
+                common_relation_methods_module.create_sig(
+                  parameters: [create_param("args", type: args_type)],
+                  return_type: "T::Enumerable[#{constant_name}]",
+                ),
+                common_relation_methods_module.create_sig(
+                  parameters: [create_param("args", type: "T.untyped")],
+                  return_type: constant_name,
+                ),
```

Comment:
This pattern doesn't seem to work
```ruby 
# test.rbi
class Foo
  sig { returns(T.untyped) }
  def foo; end

  sig { returns(T::Array[T.untyped]) }
  def bar; end
end

sig { params(args: T::Array[T.untyped]).returns(T::Enumerable[Foo]) }
sig { params(args: T.untyped).returns(Foo) }
def find(args); end
```

```ruby
# test.rb
T.reveal_type(find(Foo.new.foo)) # T::Enumerable[Foo]
T.reveal_type(find(Foo.new.bar)) # T::Enumerable[Foo]
```

First sig always takes precedence. I'm not sure why, maybe it's a sorbet bug. We could open a ticket and revert this change until its fixed.

Reply:
T.untyped argument would be matched against the first signature since it's a subtype of everything, as Jez pointed out in the linked Sorbet issue. 

Also noticed this particular issue was documented on https://sorbet.org/docs/overloads 😅 
> In the presence of untyped arguments, chances are high that the first overload is selected, which might not be desired.

Reply:
Some experimenting with the Rails console - `find(nil)` errors [here](https://github.com/rails/rails/blob/v7.1.3.2/activerecord/lib/active_record/relation/finder_methods.rb#L496-L497), `find({})` errors [here](https://github.com/rails/rails/blob/v7.1.3.2/activerecord/lib/active_record/connection_adapters/abstract/quoting.rb#L14-L28) which gives a pretty good list of types that can be passed.